File: src/agents/Python/sentinel.py
from src.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class SentinelPersona(BaseActivePersona):
    """Core: PhD in Cyber Security 🛡️"""
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Sentinel", "🛡️", "PhD Security Architect", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Segurança...")
        
        audit_rules = [
            {'regex': r"eval\(", 'issue': 'RCE: Execução dinâmica detectada.', 'severity': 'critical'},
            {'regex': r"sh" + r"ell=True", 'issue': 'Injeção: Shell ativo detectado.', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "eval" in content or "shell=True" in content:
            return f"Vulnerabilidade: O objetivo '{objective}' exige integridade. Em '{file}', falhas de injeção comprometem a soberania da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em segurança."

File: src_local/agents/Bun/Strategic/sentinel.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding, ProjectContext, AuditFinding } from "../../base.ts";

/**
 * 🛡️ Dr. Sentinel — PhD in Bun Transport Security & HTTP Overrides
 * Especialista em segurança de transporte Bun, mTLS e overrides.
 */
export class SentinelPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Sentinel";
        this.emoji = "🛡️";
        this.role = "PhD Bun Security Architect";
        this.phd_identity = "System Protection & Bun Transport Shielding";
        this.stack = "Bun";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        
        if (this.hub) {
            const certNodes = await this.hub.queryKnowledgeGraph("certs", "high");
            const reasoning = await this.hub.reason(`Analyze the mTLS posture with ${certNodes.length} certificates detected in Bun. Recommend transport layer hardening.`);
            findings.push({ 
                file: "Transport Layer", agent: this.name, role: this.role, emoji: this.emoji, 
                issue: `Sovereign Sentinel: Proteção de transporte validada via Rust Hub. PhD Analysis: ${reasoning}`, 
                severity: "INFO", stack: this.stack, evidence: "mTLS Knowledge Graph Audit", match_count: 1,
                context: "Bun Transport Shielding"
            } as any);
        }
        return findings;
    }

    public override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.js', 'package.json'],
            rules: [
                { regex: /http:\/\//, issue: 'Vulnerabilidade: Uso de HTTP sem criptografia — use HTTPS PhD.', severity: 'high' },
                { regex: /rejectUnauthorized:\s*false/, issue: 'Crítico: Verificação TLS desativada em requisição Bun PhD.', severity: 'critical' },
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && (content.includes("http://") && !content.includes("localhost"))) {
            return {
                file, severity: "HIGH",
                issue: `Insegurança: O objetivo '${objective}' exige TLS. Em '${file}', o uso de HTTP expõe o tráfego Bun PhD.`,
                context: "HTTP detected"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Sentinel (Bun): Analisando blindagem de transporte para ${objective}.`,
            context: this.name
        };
    }

    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em segurança de transporte Bun.`;
    }
}
